Scripts/RemediationScripts/Remediate-SetClassicFrontDoorMinTLSVersion.ps1

```diff
@@ -0,0 +1,1017 @@
+
+    # Includes Front Doors where Minimum required TLS version is not set to 1.2.
+    $frontDoorsEndpointWithoutReqMinTLSVersion = @()
+
+    # Includes Front Doors that were skipped during remediation. There were errors remediating them.
+    $frontDoorsSkipped = @()
+
+    $isMinTLSVersionSetOnCustomDomain = @()
+
+    $FrontDoorEndpoints = @()
+
+    $frontDoorWithoutReqMinTLSVersion = @()
+
+    Write-Host "[Step 3 of 4] Fetch all Front Door configurations"
+    Write-Host $([Constants]::SingleDashLine)
+
+    $FrontDoorResources | ForEach-Object {
+        $frontDoorResource = $_
+        $resourceId = $_.Id
+        $resourceName = $_.Name
+        #$frontendPoints = $_.FrontendEndPoints
+        $resourceGroupName= $_.Id.Split('/')[4]; 
+        try
+        {
+            Write-Host "Fetching Front Door Endpoint configuration: Resource ID: [$($resourceId)], Resource Group Name: [$($resourceGroupName)], Resource Name: [$($resourceName)]..." -ForegroundColor $([Constants]::MessageType.Info)
+            Write-Host $([Constants]::SingleDashLine)
+            
+            $FrontDoorEndpoints = Get-AzFrontDoorFrontendEndpoint -FrontDoorName $resourceName -ResourceGroupName $resourceGroupName
+        
+            if($FrontDoorEndpoints){
+            Write-Host "Front Door Configurations successfully fetched." -ForegroundColor $([Constants]::MessageType.Update)
+            Write-Host $([Constants]::SingleDashLine)
+            }
+        
+        foreach ($item in $FrontDoorEndpoints) 
+        {
+            $frontDoorResourceEP = $_
+            $minTLSVersionofEndpoint = $item.MinimumTlsVersion
+           
+           if($minTLSVersionofEndpoint.Length -gt 0)
+           {
+              if($minTLSVersionofEndpoint -lt $requiredMinTLSVersion)
+              {
+               $frontDoorsEndpointWithoutReqMinTLSVersion += $item 
+               $isMinTLSVersionSetOnCustomDomain = $false;
+              }
+           }
+           else
+           {
+               $frontDoorsEndpointWithtReqMinTLSVersion += $item
+           }   
+        }
+        }
+    catch
+        {
+            $frontDoorsSkipped += $frontDoorResource
+            Write-Host "Error fetching Front Door configuration: Resource ID: [$($resourceId)], Resource Group Name: [$($resourceGroupName)], Resource Name: [$($resourceName)]. Error: [$($_)]" -ForegroundColor $([Constants]::MessageType.Error)
+            $logResource = @{}
+            $logResource.Add("ResourceGroupName",($_.ResourceGroupName))
+            $logResource.Add("ResourceName",($_.ResourceName))
+            $logResource.Add("Reason","Encountered error while fetching Front Door configuration")    
+            $logSkippedResources += $logResource
+            Write-Host "Skipping this resource..." -ForegroundColor $([Constants]::MessageType.Warning)
+        }
+   }
+
+   foreach($item in $frontDoorsEndpointWithoutReqMinTLSVersion){
+        if (-not $frontDoorsEndpointWithoutReqMinTLSVersion)
```

Comment:
$frontDoorsEndpointWithoutReqMinTLSVersion this is a collection, we should not perform check on this, instead on individual item.

Reply:
Name is changed to noncompliantEndpoints now. and this BRS wasn't updated tarun. I fixed that already.